### PR TITLE
[BUGFIX] adaptahop frontend

### PR DESCRIPTION
**Pull request by cphyc**
## PR Summary

This fixes the support of the AdaptaHOP halo finder frontend for yt-4.

Should fix the errors appearing in #2677 .

### CODE REVIEW

File: yt/frontends/adaptahop/data_structures.py [hunk 1]
```diff
@@ -37,16 +35,18 @@
 from .fields import \
     AdaptaHOPFieldInfo
 
-class AdaptaHOPBinaryFile(HaloCatalogFile):
-    def __init__(self, ds, io, filename, file_id):
-        super(AdaptaHOPBinaryFile, self).__init__(ds, io, filename, file_id)
 
-    def _read_particle_positions(self, ptype, f=None):
-        raise NotImplementedError
+class AdaptaHOPFile(HaloCatalogFile):
```
Comment (brittonsmith) [inline yt/frontends/adaptahop/data_structures.py:39]: Can you use the base class straight away here and for `AdaptaHOPParticleIndex`?

**Comment by Xarthisius** [comment]
@cphyc could you please rebase now that #2677 was merged?

**Comment by cphyc** [comment]
> @cphyc could you please rebase now that #2677 was merged?

Done

**Comment by cphyc** [comment]
Since this PR addresses the failing tests of adaptahop that are now passing, this is ready to merge provided AppVeyor succeeds and we have enough approvals!

**Comment by munkm** [comment]
Once appveyor passes I'll be happy to merge!

**Pull request merged**